broker-daemon/block-order-worker/index.js

```diff
@@ -109,14 +109,37 @@ class BlockOrderWorker extends EventEmitter {
       this.relayer.paymentChannelNetworkService.getAddress({symbol: blockOrder.inboundSymbol})
     ])
 
-    const outboundBalanceIsSufficient = await outboundEngine.isBalanceSufficient(outboundAddress, Big(blockOrder.outboundAmount).plus(activeOutboundAmount))
+    let counterAmount
+    let outboundAmount
+    let inboundAmount
+    if (blockOrder.isMarketOrder) {
+      const { orders, depth } = await orderbook.getBestOrders({ side: blockOrder.inverseSide, depth: Big(blockOrder.baseAmount).toString() })
+      if (Big(depth).lt(blockOrder.baseAmount)) {
+        this.logger.error(`Insufficient depth`)
+        throw new Error(`Insufficient depth`)
+      }
+      counterAmount = await orderbook.getAveragePrice(orders, Big(depth))
+
+      if (blockOrder.isBid) {
+        outboundAmount = counterAmount
+        inboundAmount = blockOrder.inboundAmount
+      } else {
+        outboundAmount = blockOrder.outboundAmount
+        inboundAmount = counterAmount
+      }
+    } else {
+      outboundAmount = blockOrder.outboundAmount
+      inboundAmount = blockOrder.inboundAmount
+    }
+
+    const outboundBalanceIsSufficient = await outboundEngine.isBalanceSufficient(outboundAddress, Big(outboundAmount).plus(activeOutboundAmount))
```

Comment:
I would consider moving this sufficiency check into its own method, as `createBlockOrder` is getting pretty unwieldy